File: examples/nirfmx/nibluetoothdtm/csharp/VS2010/niBluetoothDirectTestModeRxTest/vb/NIBluetoothDirectTestModeRXTest.vb

Imports NationalInstruments.BluetoothDtm

Public Class NIBluetoothDtmRXTest

    Private dtmSession As NIBluetoothDtm
    Private dataBits As UInt16
    Private baudRate As UInt32
    Private flowControl As BluetoothDtmFlowControl
    Private stopBits As BluetoothDtmStopBits
    Private parity As BluetoothDtmParity
    Private status As Integer
    Private channelNumber As Integer
    Private packetCount As Integer
    Private visaResourceName As String

    Public Sub Run()
        Try
            InitializeVariables()
            InitializeDtmSession()
            ConfigureBlueToothDTM()
			' Add your Generation code here
            RetrieveResults()
            PrintResults()
        Catch e As Exception
            DisplayError(e.Message)
        Finally
            CloseSession()
            Console.WriteLine("Press any key to exit")
            Console.ReadKey()
        End Try
    End Sub

    Private Sub InitializeVariables()
        ' Initialize input variables 

        visaResourceName = "Com1"
        dataBits = 8
        baudRate = 115200
        flowControl = BluetoothDtmFlowControl.RtsCts
        stopBits = BluetoothDtmStopBits.StopBits1_0
        parity = BluetoothDtmParity.None
        status = 10
        channelNumber = 3
        packetCount = 0

    End Sub

    Private Sub InitializeDtmSession()
        ' Create a new BluetoothDTM Session 

        dtmSession = New NIBluetoothDtm(visaResourceName)
    End Sub

    Private Sub ConfigureBlueToothDTM()
        dtmSession.ConfigureVisaSerialSettings(dataBits, baudRate, flowControl, stopBits, parity)
        dtmSession.SetVisaTimeout(2000)
        dtmSession.HciReset(status)
        dtmSession.HciLEReceiverTest(channelNumber, status)
    End Sub

    Private Sub RetrieveResults()
        dtmSession.HciLETestEnd(packetCount, status)
        dtmSession.HciReset(status)
    End Sub

    Private Sub PrintResults()
        ' Display PacketCount

        Console.WriteLine("Number of packet count:  {0}" & vbLf, packetCount)
    End Sub



    Private Sub CloseSession()
        If dtmSession IsNot Nothing Then
            Try
                dtmSession.Close()
                dtmSession = Nothing
            Catch e As Exception
                DisplayError(e.Message)
            End Try
        End If
    End Sub

    Private Sub DisplayError(errorMessage As String)
        Console.WriteLine("ERROR: {0}", errorMessage)

    End Sub

End Class

File: examples/nirfmx/nibluetoothdtm/csharp/VS2010/niBluetoothDirectTestModeRxTest/cs/NIBluetoothDirectTestModeRXTest.cs

using System;
using NationalInstruments.BluetoothDtm;

namespace NationalInstruments.Examples.NIBluetoothDtmRXTest
{
    public class NIBluetoothDtmRXTest
    {

        NIBluetoothDtm dtmSession;
        UInt16 dataBits;
        UInt32 baudRate;
        BluetoothDtmFlowControl flowControl;
        BluetoothDtmStopBits stopBits;
        BluetoothDtmParity parity;
        int status;
        int channelNumber;
        int packetCount;
        string visaResourceName;

        public void Run()
        {
            try
            {
                InitializeVariables();
                InitializeDtmSession();
                ConfigureBlueToothDTM();
                //Add your Generation code here
                RetrieveResults();
                PrintResults();
            }
            catch (Exception e)
            {
                DisplayError(e.Message);
            }
            finally
            {
                CloseSession();
                Console.WriteLine("Press any key to exit");
                Console.ReadKey();
            }
        }

        private void InitializeVariables()
        {
            /* Initialize input variables */
            visaResourceName = "Com1";
            dataBits = 8;
            baudRate = 115200;
            flowControl = BluetoothDtmFlowControl.RtsCts;
            stopBits = BluetoothDtmStopBits.StopBits1_0;
            parity = BluetoothDtmParity.None;
            status = 10;
            channelNumber = 3;
            packetCount = 0;

        }

        private void InitializeDtmSession()
        {
            /* Create a new Bluetooth DTM Session */
            dtmSession = new NIBluetoothDtm(visaResourceName);
        }

        private void ConfigureBlueToothDTM()
        {
            dtmSession.ConfigureVisaSerialSettings(dataBits, baudRate, flowControl, stopBits, parity);
            dtmSession.SetVisaTimeout(2000);
            dtmSession.HciReset(out status);
            dtmSession.HciLEReceiverTest(channelNumber, out status);
        }

        private void RetrieveResults()
        {
            dtmSession.HciLETestEnd(out packetCount, out status);
            dtmSession.HciReset(out status);      
        }

        private void PrintResults()
        {
            /* Display PacketCount*/
            Console.WriteLine("Number of packet count:  {0}\n", packetCount);
        }

        void CloseSession()
        {
            if (dtmSession != null)
            {
                try
                {

                    dtmSession.Close();
                    dtmSession = null;
                }
                catch(Exception e)
                {
                    DisplayError(e.Message);
                }
                
            }
        }

        void DisplayError(string errorMessage)
        {
            Console.WriteLine("ERROR: {0}", errorMessage);
        }
    }
}
